packages/xo-web/src/xo-app/sr/tab-disks.js

```diff
@@ -89,13 +52,87 @@ const COLUMNS = [
     itemRenderer: vdi => formatSize(vdi.size),
     sortCriteria: vdi => vdi.size,
   },
+  {
+    name: _('vdiVms'),
+    component: connectStore(() => {
+      const getVbds = createGetObjectsOfType('VBD').pick(
+        (_, props) => props.item.$VBDs
+      )
+      const getVmIds = createSelector(getVbds, vbds => map(vbds, 'VM'))
+      const getVms = createGetObjectsOfType('VM').pick(getVmIds)
+      const getVmSnapshots = createGetObjectsOfType('VM-snapshot').pick(
+        getVmIds
+      )
+      const getAllVms = createSelector(
+        getVms,
+        getVmSnapshots,
+        (vms, vmSnapshots) => ({ ...vms, ...vmSnapshots })
+      )
+
+      return (state, props) => ({
+        vms: getAllVms(state, props),
+        vbds: getVbds(state, props),
+      })
+    })(({ vbds, vms }) => {
+      if (isEmpty(vms)) {
+        return null
+      }
+
+      return (
+        <Container>
+          {map(vbds, vbd => {
+            const vm = vms[vbd.VM]
+            const item = renderXoItem(vm)
+            const { type } = vm
+            let link
+
+            if (type === 'VM') {
+              link = `/vms/${vm.id}`
+            } else {
+              // VM-snapshot
+              const id = vm.$snapshot_of
+              link =
+                id !== undefined ? `/vms/${id}/snapshots` : '/dashboard/health'
+            }
+
+            return (
+              <Row>
+                <Col mediumSize={8}>{<Link to={link}>{item}</Link>}</Col>
+                <Col mediumSize={4}>
+                  <ButtonGroup>
+                    <ActionRowButton
+                      btnStyle='danger'
+                      disabled={vbd.attached}
+                      handler={deleteVbd}
+                      handlerParam={vbd}
+                      icon='vdi-forget'
+                      tooltip={_('vdiForget')}
```

Comment:
@olivierlambert Is "forget" still a good tooltip in the SR view? Won't the user think that the VDI will disappear from the list?

Reply:
Let's keep "forget" for now.